### PR TITLE
DRAFT: add cni cilium docs

### PR DESCRIPTION
**Pull request by kllex**
This PR adds documentation for CNIs in general and for cilium specifically.

### CODE REVIEW

File: docs/_imke/clusterlifecycle/cnichoices/index.de.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+und wie man sie im Clustererstellungsprozess auswählt.
+
+
+## Was ist eine CNI und warum brauche ich sie?
+
+CNI ist eine Abkürzung for Container Network Interface. Vereinfacht erklärt ist es eine Beschreibung der
+in Kubernetes benötigten Netzwerkfunktionalität um die Kommunkation zwischen Pods und Knoten zu realisieren.
+Da die Netzwerkumgebungen in denen Kubernetes laufen kann sehr verschieden sind haben sich die Entwickler
+von Kubernetes darauf geeinigt sich auf eine standardisierte Schnittstellenbeschreibung der benötigten
+Netzwerjfunktionalität zu beschränken und anderen die Implementierung zu überlassen. Dieser Schritt
```
Comment (qeqar) [inline docs/_imke/clusterlifecycle/cnichoices/index.de.md:21]: Typo: Netzwerkfunktionalität

File: docs/_imke/clusterlifecycle/cnichoices/index.en.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+## Cilium
+
+Cilium is a fairly new CNI which leverages eBPF instead of traditional ways to manage traffic between
+kubernetes pods and nodes. It has advanced observability features like a dedicated looking-glass addon
+called *hubble* and improved health checks. Cilium also supports network policies to allow tighter
+control of the traffic flow into or out of your cluster as well as inter-cluster traffic.
+
+If you want to utilize Cilium features to the fullest you need to set the proxy mode to *eBPF* after
+choosing the CNI (this can only be done if Konnectivity is chosen as well!). Another requirement for
+Cilium (and especially the eBFP proxy mode) is that the OS image needs to run a fairly recent kernel
```
Comment (qeqar) [inline docs/_imke/clusterlifecycle/cnichoices/index.en.md:48]: I believe that is to much information, as we only provide the right image

File: docs/_imke/clusterlifecycle/cnichoices/index.en.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+to be able to support all functionality of the CNI (which is the case with our flatcar images!).
+A compatibility matrix can be found [here](https://docs.cilium.io/en/stable/operations/system_requirements/).
+
+Cilium is under heavy development so new features as well as bugfixes are released regularly.
+
+If you are interested in a deeper understanding of the network flows inside your kubernetes cluster,
+then this CNI is for you.
+
+
+# None
```
Comment (qeqar) [inline docs/_imke/clusterlifecycle/cnichoices/index.en.md:58]: Let us patch this out?

File: docs/_imke/clusterlifecycle/controlplaneconnector/index.de.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+## Kommunikation initiiert von den Kubernetes Workern
+
+Diese Richtung der Kommunikation ist einfach realisiert da die Control-Plane ihre Dienste über den
+Kubernetes api-server anbietet, der über eine öffentliche IPv4 Adresse erreichbar ist, die den
+Kubernetes Workern wohlbekannt ist. Die Netzwerkpakete verlassen zum initiieren der Verbindung
+also das private Openstack-Subnetz über den egress-Router des Subnetzes in Richtung Internet und
+betreten dann das private Subnetz des Openstack-Tenant des Servicebetriebers über einen exponierten
+Loadbalancer.
+
+Allerdings ist es manchmal auch notwendig dass der Kubernetes API-server die Verbindung zu Diensten
```
Comment (qeqar) [inline docs/_imke/clusterlifecycle/controlplaneconnector/index.de.md:35]: Isn't that part of the next topic?

**Review by qeqar — APPROVED** [review]
/LGMT

**Review by syjabri — APPROVED** [review]

**Pull request merged**